### PR TITLE
Filter visualization tooltip list based on Index

### PR DESCRIPTION
**Pull request by corrigancd**
PR for https://github.com/sirensolutions/kibi-internal/issues/9775

I chose index rather than search as it is possible to use visualizations as popups as long as the geo_point field will be the same (from my understanding). 

Also unsure how to get the index that the visualization is linked to any other way .. 

**Manual testing:** 
1. Visualizations from multiple searches with the same index populated and selectable. "Companies with deadpooled date metric" is from deadpooled search. "Companies by Country" is from companies search
image

2. An index with no visualizations:
4399CA5B-6574-4222-A69C-5D22D3039BA5

Does a visualization always have to be

### CODE REVIEW

File: public/directives/tooltipFormatter.js [hunk 1]
```diff
@@ -33,47 +34,65 @@ define(function (require) {
...
+        fetchVisList();
 
         scope.filterSearchList = function () {
           scope.tooltipFormat.options.searchFilter = this.tooltipFormat.options.searchFilter;
           fetchSearchList();
         };
 
+        scope.filterVisList = function () {
+          scope.tooltipFormat.options.visFilter = this.tooltipFormat.options.visFilter;
+          console.log(searchService.get(this.tooltipFormat.options.visFilter));
```
Comment (szydan) [inline public/directives/tooltipFormatter.js]: remove console.log 
Next time please run
gulp lint
before making the PR
If this vis does not have such task please add it to the gulp file

File: public/directives/tooltipFormatter.js [hunk 2]
```diff
@@ -33,47 +34,65 @@ define(function (require) {
...
             });
-          });
+        };
+
+        function matchIndex(hitSavedSearchId) {
+          return _.find(scope.searchList, compare => compare.id === hitSavedSearchId);
         }
 
         function fetchVisList() {
+          //console.log(visService);
```
Comment (szydan) [inline public/directives/tooltipFormatter.js]: remove commented line

File: public/directives/tooltipFormatter.js [hunk 3]
```diff
@@ -33,47 +34,65 @@ define(function (require) {
...
-                id: hit.id
-              };
-            })
-            .value();
-          });
+            .then(hits => {
+              scope.visList = _.chain(hits.hits)
+                .filter(hit => {
+                  const visState = JSON.parse(hit.visState);
+                  return !_.includes(['enhanced_tilemap', 'tilemap', 'timelion', 'graph_browser_vis'], visState.type)
```
Comment (szydan) [inline public/directives/tooltipFormatter.js]: at the top of the  controller define the constant UNSUPPORTED_VIS_TYPES

File: package.json [hunk 4]
```diff
@@ -24,12 +24,13 @@
...
+    "babel-eslint": "^7.2.3",
     "bluebird": "^3.0.6",
-    "eslint": "^1.10.3",
-    "eslint-plugin-mocha": "1.0.0",
+    "eslint": "4.16.0",
+    "eslint-plugin-mocha": "4.11.0",
     "gulp": "^3.9.1",
-    "gulp-eslint": "^1.1.1",
+    "gulp-eslint": "^5.0.0",
     "gulp-if": "^2.0.2",
```
Comment (corrigancd) [inline package.json]: Versions are the same as graph_browser

File: package.json [hunk 5]
```diff
@@ -24,12 +24,13 @@
...
+    "eslint-config-siren": "sirensolutions/eslint-config",
+    "babel-eslint": "^7.2.3",
     "bluebird": "^3.0.6",
-    "eslint": "^1.10.3",
-    "eslint-plugin-mocha": "1.0.0",
+    "eslint": "4.16.0",
+    "eslint-plugin-mocha": "4.11.0",
     "gulp": "^3.9.1",
-    "gulp-eslint": "^1.1.1",
+    "gulp-eslint": "^5.0.0",
```
Comment (corrigancd) [inline package.json]: Versions are now the same as graph browser package.json

**Comment by corrigancd** [comment]
jenkins test it

**Comment by corrigancd** [comment]
jenkins test it

**Comment by corrigancd** [comment]
jenkins test it

**Comment by corrigancd** [comment]
jenkins test it

**Pull request merged**